go/vt/vtgate/logstats.go

```diff
@@ -121,6 +121,10 @@ func (stats *LogStats) RemoteAddrUsername() (string, string) {
 // Logf formats the log record to the given writer, either as
 // tab-separated list of logged fields or as JSON.
 func (stats *LogStats) Logf(w io.Writer, params url.Values) error {
+	if !streamlog.ShouldEmitLog(stats.SQL) {
```

Comment:
Does this include the full query or just the comments? Do you have the parsed sql in this context? I wonder if a better approach is to have a proper directive in the trailing comments.  If `stats.SQL` contains the full original query, one thing that it's not ideal about the current approach is that random queries that contain the `LogFilterTag` will sneak into the logging.

Reply:
It is the full query but without the bind variables, which is what’s logged today.

It is true that a more robust system would use properly parsed comment directives but I think this isn’t too big of a deal since users can easily come up with a filter tag that doesn’t match any of the table names / column names, and since the bind vars aren’t in the string there’s no worry about accidental matching of any column value.

Reply:
Also fwiw there isn’t an easy way to plumb the parsed query down into the logging subsystem so we’d probably need to parse again which seems overly conservative compared to the simple string search.

Reply:
Yeah, parsing again just for this does not make sense to me. 

Thanks for the clarifications.